extensions/sql-migration/src/api/utils.ts

```diff
@@ -120,3 +121,14 @@ export function filterMigrations(databaseMigrations: MigrationContext[], statusF
 	}
 	return filteredMigration;
 }
+
+export function selectDropDownIndex(dropDown: DropDownComponent, index: number): void {
+	if (index >= 0 && dropDown.values && index <= dropDown.values.length - 1) {
+		const value = dropDown.values[index];
+		dropDown.value = value as CategoryValue;
+	}
+}
+
+export function findDropDownItemIndex(dropDown: DropDownComponent, value: string): number {
+	return dropDown.values!.findIndex((v: any) => ((v as CategoryValue)?.displayName?.toLowerCase() === value?.toLowerCase()));
```

Comment:
I might have been away from javascript for a while... Could you remind me what does `!.` mean here? do you intend to write `?.`?

Reply:
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Basically it tells the compiler that despite values possibly being undefined (from the typings file) to not count that as an error. You use it when you either don't care if it's undefined (an exception being thrown is fine) or you're sure that there will be a value there when the code runs. In this case the dropDown values are being set earlier in the code so it's guaranteed to not be undefined so this is safe to do.

Reply:
I updated to add the check and handle null